src/declarativewidgets_plugin.cpp

```diff
@@ -115,6 +116,7 @@ void ExtensionpluginPlugin::registerTypes(const char *uri)
   // uncreatable core
   qmlRegisterType<QAbstractItemModel>();
   qmlRegisterType<QItemSelectionModel>();
+  qmlRegisterType<QAbstractItemDelegate>();
```

Comment:
QAbstractItemDelegate can not be constructed in C++, since it is abstract.
So this would need to be registered with qmlRegisterUncreatableType()

Reply:
I beg to differ. It's using the qmlRegisterType overload without parameters, only to make the type known to QML.
Same thing can be seen 2 lines above for QAbstractItemModel.
Should this really be handled with qmlRegisterUncreatableType?

Reply:
Hmm, good point.
I guess we should change that to uncreatable type registration, in a future clean up commit, and for the other uncreatables as well